### PR TITLE
Prepare for 0.12.0 Release

### PR DESCRIPTION
**Pull request by mbaijal**
## Description ##
* Bump up version to 0.12.0
* Update NEWS for v0.12.0 and link to release notes

### Essentials ###
- [ ] Passed code style checking (`make lint`)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] To my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Updated all the necessary files with version 0.12.0

## Comments ##
Coming up in In another PR before tagging the RC -  
Updated README.md
Updated NEWS.md if needed

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -1,5 +1,30 @@
 MXNet Change Log
 ================
+## 0.12.0
+### - New Features - Sparse Tensor Support
```
Comment (piiswrong) [inline NEWS.md]: Don't need the leading -

File: NEWS.md [hunk 2]
```diff
@@ -1,5 +1,30 @@
...
+  - `gluon.Trainer` now allows reading and setting learning rate with `trainer.learning_rate` property.
+  - Added `mx.autograd.grad` and experimental second order gradient support (though most operators don't support second order gradient yet)
+  - Added `ConvLSTM` etc to `gluon.contrib`
+  - Autograd now supports cross-device graphs. Use `x.copyto(mx.gpu(i))` and `x.copyto(mx.cpu())` to do computation on multiple devices.
+### - Other New Features
+  - Random number generators in `mx.nd.random.*` and `mx.sym.random.*` now supports both CPU and GPU
+  - `NDArray` and `Symbol` now supports "fluent" methods. You can now use `x.exp()` etc instead of `mx.nd.exp(x)` or `mx.sym.exp(x)`
+  - Added `mx.rtc.CudaModule` for writing and running CUDA kernels from python
+  - Added `multi_precision` option to optimizer for easier float16 training
+### - Performance
```
Comment (piiswrong) [inline NEWS.md]: Why are two other performance improvements removed?

File: NEWS.md [hunk 3]
```diff
@@ -1,5 +1,30 @@
...
+### - New Features - Sparse Tensor Support
+  - Added comprehensive support for sparse matrices. See help on `mx.sym.sparse` and `mx.nd.sparse` for more info.
+  - For more information see [full release notes](https://cwiki.apache.org/confluence/display/MXNET/MXNet+0.12.0+Release+Notes)
+### - New Features - Autograd and Gluon
+  - New loss functions added - `SigmoidBinaryCrossEntropyLoss`, `CTCLoss`, `HuberLoss`, `HingeLoss`, `SquaredHingeLoss`, `LogisticLoss`, `TripletLoss`
+  - `gluon.Trainer` now allows reading and setting learning rate with `trainer.learning_rate` property.
+  - Added `mx.autograd.grad` and experimental second order gradient support (though most operators don't support second order gradient yet)
+  - Added `ConvLSTM` etc to `gluon.contrib`
+  - Autograd now supports cross-device graphs. Use `x.copyto(mx.gpu(i))` and `x.copyto(mx.cpu())` to do computation on multiple devices.
+### - Other New Features
```
Comment (piiswrong) [inline NEWS.md]: Also list `limited support for fancy indexing, full support coming soon in next release. Checkout master to get a preview`

File: NEWS.md (hunk 1 above)
Comment (mbaijal) [reply NEWS.md]: Done

File: NEWS.md (hunk 3 above)
Comment (mbaijal) [reply NEWS.md]: Done

File: NEWS.md (hunk 2 above)
Comment (mbaijal) [reply NEWS.md]: Added 1 back as discussed - " Enabled JIT compilation. Autograd and Gluon hybridize now use less memory and has faster speed. Performance is almost the same with old symbolic style code"

File: docs/build_version_doc/build_all_version.sh [hunk 4]
```diff
@@ -21,7 +21,7 @@
 # Built files are stored in $built
 # Version numbers are stored in $tag_list.
 # Version numbers are ordered from latest to old and final one is master.
-tag_list="0.11.0.rc3 master"
+tag_list="0.12.0 0.11.0 master"
```
Comment (mbaijal) [inline docs/build_version_doc/build_all_version.sh]: @sandeep-krishnamurthy @kevinthesun 
Should this change be made now or after the RC is final?

File: NEWS.md [hunk 5]
```diff
@@ -1,11 +1,38 @@
 MXNet Change Log
 ================
+## 0.12.0
+### New Features - Sparse Tensor Support
+  - Added comprehensive support for sparse matrices. See help on `mx.sym.sparse` and `mx.nd.sparse` for more info.
+  - Limited support for fancy indexing. x[idx_arr0, idx_arr1, ..., idx_arrn] is now supported. Full support coming soon in next release. Checkout master to get a preview.
```
Comment (eric-haibin-lin) [inline NEWS.md]: fancy indexing is for normal NDArray

Comment (mbaijal) [reply NEWS.md]: @eric-haibin-lin Can you please edit the line and let me know how you want to rephrase it?

File: NEWS.md [hunk 6]
```diff
@@ -1,11 +1,38 @@
...
+  - Autograd now supports cross-device graphs. Use `x.copyto(mx.gpu(i))` and `x.copyto(mx.cpu())` to do computation on multiple devices.
+### Other New Features
+  - Random number generators in `mx.nd.random.*` and `mx.sym.random.*` now supports both CPU and GPU
+  - `NDArray` and `Symbol` now supports "fluent" methods. You can now use `x.exp()` etc instead of `mx.nd.exp(x)` or `mx.sym.exp(x)`
+  - Added `mx.rtc.CudaModule` for writing and running CUDA kernels from python
+  - Added `multi_precision` option to optimizer for easier float16 training
+### Performance
+  - Enabled JIT compilation. Autograd and Gluon hybridize now use less memory and has faster speed. Performance is almost the same with old symbolic style code.
+  - Full support for NVidia Volta GPU Architecture and Cuda 9. Training is up to 3.5x faster than Pascal when using float16.
+### - API Changes
```
Comment (eric-haibin-lin) [inline NEWS.md]: why is there a dash before API Changes?

Comment (mbaijal) [reply NEWS.md]: Good catch 👍

File: NEWS.md [hunk 7]
```diff
@@ -1,11 +1,38 @@
 MXNet Change Log
 ================
+## 0.12.0
+### New Features - Sparse Tensor Support
+  - Added comprehensive support for sparse matrices. See help on `mx.sym.sparse` and `mx.nd.sparse` for more info.
```
Comment (eric-haibin-lin) [inline NEWS.md]: - Added limited cpu support for two sparse formats for `Symbol` and `NDArray` - `CSRNDArray` and `RowSparseNDArray`
- Added a sparse dot product operator and many element-wise sparse operators
- Added a data iterator for sparse data input - `LibSVMIter`
- Added three optimizers for sparse gradient updates: `Ftrl`, `SGD` and `Adam`
- Added `push` and `row_sparse_pull` with `RowSparseNDArray` in distributed kvstore

File: NEWS.md (hunk 5 above)
Comment (eric-haibin-lin) [reply NEWS.md]: I am just saying that this doesn't belong to sparse tensor

File: NEWS.md (hunk 7 above)
Comment (mbaijal) [reply NEWS.md]: Should I replace the existing point or add subpoints?

File: NEWS.md (hunk 5 above)
Comment (mbaijal) [reply NEWS.md]: Moved to other features. Done.

File: NEWS.md (hunk 7 above)
Comment (eric-haibin-lin) [reply NEWS.md]: Yeah please replace the existing point. `comprehensive support ` is misleading since we dont have gpu support. Thanks!

Comment (mbaijal) [reply NEWS.md]: done

Comment (eric-haibin-lin) [reply NEWS.md]: Ah, all the formats are gone.. :(

File: NEWS.md [hunk 8]
```diff
@@ -1,11 +1,42 @@
 MXNet Change Log
 ================
+## 0.12.0
+### New Features - Sparse Tensor Support
+  - Added limited cpu support for two sparse formats for Symbol and NDArray - CSRNDArray and RowSparseNDArray
+  - Added a sparse dot product operator and many element-wise sparse operators
+  - Added a data iterator for sparse data input - LibSVMIter
+  - Added three optimizers for sparse gradient updates: Ftrl, SGD and Adam
+  - Added push and row_sparse_pull with RowSparseNDArray in distributed kvstore
+  - For more information see [full release notes](https://cwiki.apache.org/confluence/display/MXNET/MXNet+0.12.0+Release+Notes)
```
Comment (eric-haibin-lin) [inline NEWS.md]: +  - For more information see [full release notes](https://cwiki.apache.org/confluence/display/MXNET/MXNet+0.12.0+Release+Notes)
Why is this put under sparse? Should we move it to the end?
